astropy/coordinates/baseframe.py

```diff
@@ -1765,6 +1766,49 @@ def __eq__(self, value):
     def __ne__(self, value):
         return np.logical_not(self == value)
 
+    def _prepare_unit_sphere_coords(
+        self, other: BaseCoordinateFrame | SkyCoord
+    ) -> tuple[Longitude, Latitude, Longitude, Latitude]:
+        self_sph = self.represent_as(r.UnitSphericalRepresentation)
+        other_sph = (
+            getattr(other, "frame", other)
+            .transform_to(self)
+            .represent_as(r.UnitSphericalRepresentation)
+        )
+        return self_sph.lon, self_sph.lat, other_sph.lon, other_sph.lat
+
+    def position_angle(self, other: BaseCoordinateFrame | SkyCoord) -> Angle:
+        """Compute the on-sky position angle to another coordinate.
+
+        Parameters
+        ----------
+        other : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+            The other coordinate to compute the position angle to.  It is
+            treated as the "head" of the vector of the position angle.
+
+        Returns
+        -------
+        `~astropy.coordinates.Angle`
+            The (positive) position angle of the vector pointing from ``self``
+            to ``other``.  If either ``self`` or ``other`` contain arrays, this
```

Comment:
let's add ", measured East from North" to give people new to position angles a clue as to what to expect (though the examples help!)